src/main/java/uk/gov/verifiablelog/merkletree/InMemoryPowOfTwo.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.verifiablelog.merkletree;
+
+
+import java.util.TreeMap;
+
+public class InMemoryPowOfTwo implements MemoizationStore {
+
+    private TreeMap<Integer, TreeMap<Integer, byte[]>> data;
+
+    public InMemoryPowOfTwo() {
+        data = new TreeMap<>();
+    }
+
+
+    @Override
+    public void put(Integer start, Integer size, byte[] value) {
+        if (((start == 0) || (Integer.bitCount(start) == 1)) &&
+                ((size == 0) || (Integer.bitCount(size) == 1))) {
```

Comment:
maybe could use `<= 1` rather than `== 1` to include the zero case? is that clearer?

also is it worth extracting a method with an intension-revealing name such as `isZeroOrPowerOfTwo`?